### ISSUE TITLE
Could we have R2 as a metrics choice when compiling a regression model? [feature request]

### ISSUE THREAD

**Issue by UnixJunkie** [keras-team/keras#14090]
Dear keras hackers,

I am using the R interface to keras, if that matters.

In my field, people use the coefficient of determination (R2) to assess the quality of a regression model (Cf. https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html
for explanations).

And, if R2 is made available, I guess that the loss we would want to monitor while training
a model should be (1 - R2).

So, this feature request is about a new loss function (1 - R2), and the associated regression metric (R2).

Thanks a lot,
F.

**Comment by UnixJunkie** [comment]
cc @dabasajay @tobigithub

**Comment by goelharshit562** [comment]
R2 behaviour is unexpected because sometimes a good model can have a low R2 value. On the other hand, a biased model can have a high R2 value.

**Comment by tobigithub** [comment]
@UnixJunkie I think that would be fine, implemented via custom functions or otherwise,
frameworks should be open to every solution and only benchmarking would reveal any good/bad performance. RMSE usually performs better that R2, but why not, its up to the user. Similar to Eureqa's genetic programming features, the program really does not care, there is no prejudice, its all based on science, all options open, people can try whatever they think is best. Its a framework!

**Comment by sushreebarsa** [comment]
@UnixJunkie  Moving this issue to closed status as there has been no recent activity, in case you still face the error please create a new issue.Thanks!
